### PR TITLE
Interactive functionality added

### PR DESCRIPTION
**Pull request by karanJ2212**
in this milestone I have added below functionality
1- click on checkbox, task will be striked and marked completed. property completed: true
2-clear all the completed task with clear all button
3- index of the task will be updated accordingly

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,37 +1,141 @@
...
+// class toDoTask {
+//   constructor(desc, index, completed = false) {
+//     this.desc = desc;
+//     this.index = index;
+//     this.completed = completed;
+//   }
+// }
+
+//start ui class
+
```
Comment (elisha2kyakpo1) [inline src/index.js:16]: - [ ] Kindly remove all the commented code to clean up the codebase.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @karanJ2212, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look excellent and professional.
✔️ No linter errors.
✔️ You have descriptive commit messages.
✔️ Descriptive readme file.
✔️ The app works as expected👏.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [ ] The readme file shows the users should `cd awesome_book` when this is a todo list. 

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (karanJ2212) [reply src/index.js:16]: done

**Review by omarsalem7 — APPROVED** [review]
Hi @karanJ2212 😊, 

# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
- There is no linter error present ✔️
- The Pull Request has a proper description ✔️
- The project has met all the technical aspects you needed for this project. Keep rocking!  ✔️


Cheers and Happy coding!👏👏👏

## Optional suggestions

- [ ] [OPTIONAL] The commit message should be written in the imperative tense and the first letter should also be capitalized. [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)
- [ ] [OPTIONAL] It will be better if you add a license MIT link to your project to make it professional.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**